e2e/setup.go

```diff
@@ -324,11 +332,32 @@ func createS3Bucket(bucketName string) {
 	}
 }
 
-func RandString(n int) string {
+// StringOrByte ... constraint that generic type is either "string" or "[]byte"
+type StringOrByte interface {
+	string | []byte
+}
+
+func randStr(n int) string {
 	var letterRunes = []rune("abcdefghijklmnopqrstuvwxyz")
 	b := make([]rune, n)
 	for i := range b {
 		b[i] = letterRunes[rand.Intn(len(letterRunes))]
 	}
 	return string(b)
 }
+
+// Rand generates either a random string or byte slice depending on the type T
+func Rand[T StringOrByte](length int) T {
+	str := randStr(length)
+
+	// Use type switch to return the correct type
+	var result any
+	switch any(new(T)).(type) {
+	case *string:
+		result = str
+	case *[]byte:
+		result = []byte(str)
+	}
+
+	return result.(T)
+}
```

Comment:
Think this is overcomplicating things and not necessary. Generics typically when behavior is the same, but types are different. Here we just want 2 different things. Why not just have 2 separate functions RandByte and RandString? Super straightforward, follows stdlib pattern. At your call site if I see the generic I'm not sure what's happening so I need to click through and then parse this complex function, for something that should be completely trivial and uninteresting.

Reply:
[2d3559f](https://github.com/Layr-Labs/eigenda-proxy/pull/182/commits/2d3559f260ad130dbefd835f3ab13407a5494381)